contribs/gnodev/pkg/browser/model.go

```diff
@@ -0,0 +1,572 @@
+	return func() tea.Msg {
+		res, err := m.client.Call(path, command)
+		return execCommandMsg{res, err}
+	}
+}
+
+func (m *model) ExtendCommandInput() bool {
+	if !m.commandInput.Focused() {
+		return false
+	}
+
+	if item, ok := m.listFuncs.SelectedItem().(itemFunc); ok {
+		var value string
+		if len(item.Params) > 0 {
+			value = item.Title() + "("
+		} else {
+			value = item.Title() + "()"
+		}
+
+		currentValue := m.commandInput.Value()
+		if len(value) > len(currentValue) && strings.HasPrefix(value, currentValue) {
+			m.commandInput.SetValue(value)
+			return true
+		}
+
+		// Put cursor at the end
+		m.commandInput.CursorEnd()
+	}
+
+	return false
+}
+
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
+
+	switch msg := msg.(type) {
+	case fetchRealmMsg:
+		if msg.realmPath != "" {
+			cmd = m.moveToRealm(msg.realmPath)
+		} else {
+			path := m.getCurrentPath()
+			m.logger.Info("rendering realm", "path", path)
+			cmd = m.RenderUpdate(m.getCurrentPath())
+		}
+
+		return m, tea.Sequence(m.taskLoader.Add(1), cmd)
+
+	case execCommandRequestMsg:
+		m.logger.Info("requesting command", "path", msg.Path, "cmd", msg.Command)
+		cmd = m.ExecCommand(msg.Path, msg.Command)
+		return m, tea.Sequence(m.taskLoader.Add(1), cmd)
+
+	case execCommandMsg:
+		m.taskLoader.Done()
+
+		if msg.Error != nil {
+			m.logger.Warn("command exec", "error", msg.Error)
+			content := fmt.Sprintf(
+				"%s\n\npress [enter] to dismiss",
+				warpLine(msg.Error.Error(), m.viewport.Width),
+			)
+			m.viewport.SetContent(content)
+			m.messageDisplay = true
+		} else {
+			content := string(msg.Response)
+			if strings.TrimSpace(content) == "" {
+				cmd = RefreshRealm()
+				m.messageDisplay = false
+			} else {
+				m.viewport.SetContent(fmt.Sprintf("%s\n\npress [enter] to dismiss", content))
+				m.messageDisplay = true
+			}
+		}
+
+		return m, cmd
+
+	case renderUpdateMsg:
+		m.taskLoader.Done()
+
+		var content string
+		if err := msg.Error; err != nil {
+			m.logger.Warn("render", "error", msg.Error)
+			// Write error to the frame
+			content = fmt.Sprintf("ERROR: %s", err.Error())
+		} else {
+			content = string(m.findAndMarkURLs(msg.Render))
+		}
+
+		if len(msg.Funcs) > 0 {
+			items := make([]list.Item, 0, len(msg.Funcs))
+			for _, fun := range msg.Funcs {
+				if fun.FuncName != "Render" {
+					items = append(items, itemFunc(fun))
+				}
+			}
+			m.listFuncs.SetItems(items)
+			m.listFuncs.FilterItems(m.commandInput.Value())
+
+			// Update funcs list
+			m.listFuncs.Title = m.urlInput.Value()
+			m.listFuncs.SetSize(m.viewport.Width, 7)
+		}
+
+		m.viewport.SetContent(content)
+
+		return m, cmd
+	case SpinnerTickMsg:
+		if m.taskLoader.Active() {
+			m.taskLoader, cmd = m.taskLoader.Update(msg)
+		}
+
+	case tea.MouseMsg:
+		cmd = m.updateMouse(msg)
+
+		// Fallback on viewport
+		if cmd == nil {
+			m.viewport, cmd = m.viewport.Update(msg)
+		}
+
+		return m, cmd
+
```

Comment:
Only an Emacs/Vim user can understand why ahahah 😛 

<details>
  <summary>spoiler</summary>
	You will find this more with Emacs users, I think. Since you don't use a mouse, you navigate through code 	using arrow keys (or `h, j, k, l`) and you move either line by line or paragraph by paragraph. That's why I put so much space in my code; it naturally helps to navigate. I also think it's a good thing because this natural spacing also helps with readability :)
</details>